server/channels/jobs/delete_empty_drafts_migration/delete_empty_drafts_migration.go

```diff
@@ -0,0 +1,222 @@
+		name:      JobName,
+		stop:      make(chan bool, 1),
+		stopped:   make(chan bool, 1),
+		jobs:      make(chan model.Job),
+	}
+	return worker
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) Run() {
+	mlog.Debug("Worker started", mlog.String("worker", worker.name))
+	// We have to re-assign the stop channel again, because
+	// it might happen that the job was restarted due to a config change.
+	worker.stop = make(chan bool, 1)
+
+	defer func() {
+		mlog.Debug("Worker finished", mlog.String("worker", worker.name))
+		worker.stopped <- true
+	}()
+
+	for {
+		select {
+		case <-worker.stop:
+			mlog.Debug("Worker received stop signal", mlog.String("worker", worker.name))
+			return
+		case job := <-worker.jobs:
+			mlog.Debug("Worker received a new candidate job.", mlog.String("worker", worker.name))
+			worker.DoJob(&job)
+		}
+	}
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) Stop() {
+	mlog.Debug("Worker stopping", mlog.String("worker", worker.name))
+	close(worker.stop)
+	<-worker.stopped
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) JobChannel() chan<- model.Job {
+	return worker.jobs
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) IsEnabled(_ *model.Config) bool {
+	return true
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) getJobMetadata(job *model.Job, key string) (int64, *model.AppError) {
+	countStr := job.Data[key]
+	count := int64(0)
+	var err error
+	if countStr != "" {
+		count, err = strconv.ParseInt(countStr, 10, 64)
+		if err != nil {
+			return 0, model.NewAppError("getJobMetadata", model.NoTranslation, nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+	}
+	return count, nil
+}
+
+func (worker *DeleteEmptyDraftsMigrationWorker) DoJob(job *model.Job) {
+	defer worker.jobServer.HandleJobPanic(job)
+
+	if claimed, err := worker.jobServer.ClaimJob(job); err != nil {
+		mlog.Warn("DeleteEmptyDraftsMigrationWorker experienced an error while trying to claim job",
+			mlog.String("worker", worker.name),
+			mlog.String("job_id", job.Id),
+			mlog.Err(err))
+		return
+	} else if !claimed {
+		return
+	}
+
+	var appErr *model.AppError
+	// We get the job again because ClaimJob changes the job status.
+	job, appErr = worker.jobServer.GetJob(job.Id)
+	if appErr != nil {
+		mlog.Error("DeleteEmptyDraftsMigrationWorker: job execution error", mlog.String("worker", worker.name), mlog.String("job_id", job.Id), mlog.Err(appErr))
+		worker.setJobError(job, appErr)
+		return
+	}
+
+	// Wait for all clusters to finish DB migration
+	clusterStatus := worker.app.GetClusterStatus()
+	for i := 1; i < len(clusterStatus); i++ {
+		if clusterStatus[i].SchemaVersion != clusterStatus[0].SchemaVersion {
+			// Just wait for the next loop
+			worker.jobServer.SetJobPending(job)
+			return
+		}
+	}
+
+	// Check if there is metadata for that job.
+	// If there isn't, it will be empty by default, which is the right value.
+	userID := job.Data["user_id"]
+	createAt, appErr := worker.getJobMetadata(job, "create_at")
```

Comment:
`getJobMetadata` is just a helper function for converting the data to a `string`, it's a bit misnamed.
I think for this job I can remove it since we're just using it in the one place, and it is confusing.